part/map.go

```diff
@@ -383,3 +383,82 @@ func (m *Map[K, V]) UnmarshalYAML(value *yaml.Node) error {
 	m.tree = txn.CommitOnly()
 	return nil
 }
+
+func (m Map[K, V]) Txn() MapTxn[K, V] {
+	m.ensureTree()
+	txn := m.tree.Txn()
+	if m.singleton != nil {
+		txn.Insert(m.keyToBytes(m.singleton.Key), mapKVPair[K, V]{m.singleton.Key, m.singleton.Value})
+	}
+	bytesFromKey := m.bytesFromKeyFunc
+	if bytesFromKey == nil {
+		bytesFromKey = lookupKeyType[K]()
+	}
+	return MapTxn[K, V]{
+		bytesFromKeyFunc: bytesFromKey,
+		txn:              txn,
+	}
+}
+
+// MapTxn is a write transaction for efficiently doing multiple
+// modifications to a map.
+type MapTxn[K, V any] struct {
+	bytesFromKeyFunc func(K) []byte
+	txn              *Txn[mapKVPair[K, V]]
+}
+
+func (txn MapTxn[K, V]) Commit() (m Map[K, V]) {
```

Comment:
I guess Abort is just let GC collect everything not committed?

Reply:
Yep